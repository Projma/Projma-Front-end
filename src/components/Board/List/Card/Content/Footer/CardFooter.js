import CheckBoxOutlinedIcon from "@mui/icons-material/CheckBoxOutlined";
import ChatBubbleIcon from "@mui/icons-material/ChatBubble";
import Avatar from "@mui/material/Avatar";
import AvatarGroup from "@mui/material/AvatarGroup";
import AttachFileIcon from "@mui/icons-material/AttachFile";
import Tooltip, { tooltipClasses } from "@mui/material/Tooltip";
import { convertNumberToPersian } from "../../../../../../utilities/helpers";

const CardFooter = ({
  doers,
  attachments_num,
  checklists_num,
  checked_checklists_num,
  comments_num,
}) => {
  return (
    <>
      <div className="card_footer">
        <div className="card_card-avatar">
          {doers !== [] && (
            <AvatarGroup
              max={5}
              spacing="-1"
              sx={{ direction: "ltr", border: "none" }}
              className="card_avatar-container"
            >
              {doers.map((x) => (
                <Tooltip title={x.first_name + " " + x.last_name}>
                  <Avatar
                    key={crypto.randomUUID()}
                    alt={x.first_name + " " + x.last_name}
                    src={x.profile_pic !== null ? x.profile_pic : "none"}
                    {...stringAvatar(x.first_name + " " + x.last_name)}
                    className="card_avatar-profile-picture"
                  />
                </Tooltip>
              ))}
            </AvatarGroup>
          )}
        </div>
        <div className="card_footer-icon">
          {attachments_num !== 0 && attachments_num !== undefined && (
            <div className="card_icon-container">
              <AttachFileIcon className="card_default-footer-icon" />
              <p className="card_icon-info">
                {convertNumberToPersian(attachments_num)}
              </p>
            </div>
          )}
          {checklists_num !== 0 && checklists_num !== undefined && (
            <div>
              {checked_checklists_num === checklists_num ? (
                <div className="card_icon-container">
                  <CheckBoxOutlinedIcon className="card_default-footer-icon card_checklist-finish" />
                  <p className="card_icon-info ">
                    {convertNumberToPersian(checked_checklists_num)} /
                    {convertNumberToPersian(checklists_num)}
                  </p>
                </div>
              ) : (
                <div className="card_icon-container">
                  <CheckBoxOutlinedIcon className="card_default-footer-icon" />
                  <p className="card_icon-info">
                    {convertNumberToPersian(checked_checklists_num)}/
                    {convertNumberToPersian(checklists_num)}
                  </p>
                </div>
              )}
            </div>
          )}
          {comments_num !== 0 && comments_num !== undefined && (
            <div className="card_icon-container">
              <ChatBubbleIcon className="card_default-footer-icon" />
              <p className="card_icon-info">
                {convertNumberToPersian(comments_num)}
              </p>
            </div>
          )}
        </div>
      </div>
    </>
  );
};

function stringToColor(string) {
  let hash = 0;
  let i;

  /* eslint-disable no-bitwise */
  for (i = 0; i < string.length; i += 1) {
    hash = string.charCodeAt(i) + ((hash << 5) - hash);
  }

  let color = "#";

  for (i = 0; i < 3; i += 1) {
    const value = (hash >> (i * 8)) & 0xff;
    color += `00${value.toString(16)}`.slice(-2);
  }
  /* eslint-enable no-bitwise */

  return color;
}

function stringAvatar(name) {
  return {
    sx: {
      bgcolor: stringToColor(name),
    },
    children: `${name.split(" ")[0][0].toUpperCase()}${name
      .split(" ")[1][0]
      .toUpperCase()}`,
  };
}

export default CardFooter;
