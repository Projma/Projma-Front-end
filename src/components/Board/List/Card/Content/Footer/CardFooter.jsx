import CheckBoxOutlinedIcon from "@mui/icons-material/CheckBoxOutlined";
import ChatBubbleIcon from "@mui/icons-material/ChatBubble";
import Avatar from "@mui/material/Avatar";
import AvatarGroup from "@mui/material/AvatarGroup";
import AttachFileIcon from "@mui/icons-material/AttachFile";
import Tooltip, { tooltipClasses } from "@mui/material/Tooltip";
import { convertNumberToPersian } from "../../../../../../utilities/helpers";
import useTheme from "../../../../../../hooks/useTheme";

const CardFooter = ({
  doers,
  attachments_num,
  checklists_num,
  checked_checklists_num,
  comments_num,
}) => {
  const {theme, getColor} = useTheme();
  return (
    <>
      {(attachments_num !== 0 ||
        checklists_num !== 0 ||
        comments_num !== 0 ||
        doers.length !== 0) && (
        <div className="card_footer">
          {doers.length !== 0 && (
            <div className="card_card-avatar">
              <AvatarGroup
                max={5}
                sx={{ direction: "ltr", border: "none" }}
                className="card_avatar-container"
              >
                {doers.map((x) => (
                  <Tooltip title={x.first_name + " " + x.last_name}>
                    <Avatar
                      key={crypto.randomUUID()}
                      alt={x.first_name + " " + x.last_name}
                      src={x.profile_pic !== null ? x.profile_pic : "none"}
                      sx={{color: getColor(theme.secondary), backgroundColor: theme.secondary}}
                      className="card_avatar-profile-picture"
                    >
                      {(x.first_name[0] + x.last_name[0]).toUpperCase()}
                    </Avatar>
                  </Tooltip>
                ))}
              </AvatarGroup>
            </div>
          )}
          {(attachments_num !== 0 ||
            checklists_num !== 0 ||
            comments_num !== 0) && (
            <div className="card_footer-icon">
              {attachments_num !== 0 && attachments_num !== undefined && (
                <div className="card_icon-container">
                  <AttachFileIcon className="card_default-footer-icon" />
                  <p className="card_icon-info">
                    {convertNumberToPersian(attachments_num)}
                  </p>
                </div>
              )}
              {checklists_num !== 0 && checklists_num !== undefined && (
                <div>
                  {checked_checklists_num === checklists_num ? (
                    <div className="card_icon-container">
                      <CheckBoxOutlinedIcon className="card_default-footer-icon card_checklist-finish" />
                      <p className="card_icon-info ">
                        {convertNumberToPersian(checked_checklists_num)} /
                        {convertNumberToPersian(checklists_num)}
                      </p>
                    </div>
                  ) : (
                    <div className="card_icon-container">
                      <CheckBoxOutlinedIcon className="card_default-footer-icon" />
                      <p className="card_icon-info">
                        {convertNumberToPersian(checked_checklists_num)}/
                        {convertNumberToPersian(checklists_num)}
                      </p>
                    </div>
                  )}
                </div>
              )}
              {comments_num !== 0 && comments_num !== undefined && (
                <div className="card_icon-container">
                  <ChatBubbleIcon className="card_default-footer-icon" />
                  <p className="card_icon-info">
                    {convertNumberToPersian(comments_num)}
                  </p>
                </div>
              )}
            </div>
          )}
        </div>
      )}
    </>
  );
};

export default CardFooter;
