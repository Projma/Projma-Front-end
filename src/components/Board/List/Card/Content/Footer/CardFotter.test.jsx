import React from 'react';
import { render } from '@testing-library/react';
import CardFooter from './CardFooter';

describe('CardFooter', () => {
  vi.mock("../../../../../../hooks/useTheme.jsx", () => ({
    __esModule: true,
    default: () => ({
      theme: {
        name: "sun",
        mainBg: "#e5e5e5",
        minorBg: "#fff",
        secondary: "#f8981c",
        tertiary: "#f47922",
        hover: "#f4792280",
        primary: "#fdb713",
      },
      getColor: (bgColor) => "#000000",
    }),
  }));
  
  it('renders correctly', () => {
    const doers = [
      {
        first_name: 'John',
        last_name: 'Doe',
        profile_pic: 'https://example.com/john.jpg'
      },
      {
        first_name: 'Jane',
        last_name: 'Doe',
        profile_pic: 'https://example.com/jane.jpg'
      }
    ];
    const attachments_num = 2;
    const checklists_num = 3;
    const checked_checklists_num = 1;
    const comments_num = 4;

    const { container } = render(
      <CardFooter
        doers={doers}
        attachments_num={attachments_num}
        checklists_num={checklists_num}
        checked_checklists_num={checked_checklists_num}
        comments_num={comments_num}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
