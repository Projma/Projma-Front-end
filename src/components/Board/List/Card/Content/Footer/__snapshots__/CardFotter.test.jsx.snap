// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardFooter > renders correctly 1`] = `
<div>
  <div
    class="card_footer"
  >
    <div
      class="card_card-avatar"
    >
      <div
        class="MuiAvatarGroup-root card_avatar-container css-1a5pyhb-MuiAvatarGroup-root"
      >
        <div
          aria-label="Jane Doe"
          class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar card_avatar-profile-picture css-1ixhonp-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <img
            alt="Jane Doe"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://example.com/jane.jpg"
          />
        </div>
        <div
          aria-label="John Doe"
          class="MuiAvatar-root MuiAvatar-circular MuiAvatarGroup-avatar card_avatar-profile-picture css-1ixhonp-MuiAvatar-root"
          data-mui-internal-clone-element="true"
        >
          <img
            alt="John Doe"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://example.com/john.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="card_footer-icon"
    >
      <div
        class="card_icon-container"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium card_default-footer-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="AttachFileIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
          />
        </svg>
        <p
          class="card_icon-info"
        >
          ۲
        </p>
      </div>
      <div>
        <div
          class="card_icon-container"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium card_default-footer-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V5h14v14zM17.99 9l-1.41-1.42-6.59 6.59-2.58-2.57-1.42 1.41 4 3.99z"
            />
          </svg>
          <p
            class="card_icon-info"
          >
            ۱
            /
            ۳
          </p>
        </div>
      </div>
      <div
        class="card_icon-container"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium card_default-footer-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="ChatBubbleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
          />
        </svg>
        <p
          class="card_icon-info"
        >
          ۴
        </p>
      </div>
    </div>
  </div>
</div>
`;
